contracts/Kasu.sol

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.0;
+
+import "hardhat/console.sol";
+
+contract Kasu {
+    uint public listingsCount; // keep track of number of listings
+
+    constructor() public {
+        listingsCount = 0;
+    }
+
+    enum RentalStatus { 
+        Available, 
+        Unavailable 
+    }
+
+    // Rent will be calcualted as collateral required (1ETH) * Daily interest rate (1%) * duration (5 days) 
+    // 1ETH * 1% daily interest rate * 5 days = 0.05ETH total
+    struct Listing {
+        uint256 tokenId; // tokenId from ERC 721 https://ethereum.org/en/developers/docs/standards/tokens/erc-721/
+        address tokenAddress;
+        address payable lenderAddress; // lenderAddress
+        uint16 duration; // duration of the loan in days
+        uint16 dailyInterestRate; // daily interest rate
+        uint256 collateralRequired; // collateral required
+        RentalStatus rentalStatus; // status of rental 
+    }
+
+    struct Rental {
+        address payable borrowerAddress;
+        uint16 rentDuration; // current rent duration
+        uint256 rentedAt; // timestamp at which the NFT was rented
+    }
```

Comment:
I split up Listing and Rental. We can also combine them into one listing but I thought that was confusing because you have lender and borrower specific attributes. In the future we can create a mapping of token id ==> RentalContract but this requires more storage. 

```
struct RentalContract {
    Listing listing;
    Rental rental;
}
```

Reply:
I'm ok either way, but if this is just for clarity you can just embed the Rental struct inside Listing (so you don't have to separately maintain a rentalStatus - it's rented if `rentedAt != 0`).
Mapping might be a bit tricky since an NFT is identified by both tokenAddress and tokenId. It would be a lot easier if each Listing has a listingId (also easy for subsequent operations).

Reply:
yeah I'll add a `Rental` field in the `Listing` struct. If NFT is identified both tokenAddress + tokenId, I'll add a `listing_id` + a mapping of listing_id => Listing